### PR TITLE
Fix: #1344 Refactor all occurences of Hamcrest assertions with AssertJ equivalents

### PR DESCRIPTION
**Pull request by anurag-rajawat**
## Description

Fixes: #1344 
Refactored all the occurrences of hamcrest assertions in the entire jkube repository with the Assertj equivalent

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by anurag-rajawat** [comment]
@rohanKanojia in my local environment the build is successful, but here it got failed, why?
Also, I'm attaching a screenshot for your reference


Screenshot from 2022-03-19 13-31-02
.

**Comment by rohanKanojia** [comment]
Please fix applicable Sonar code smells

**Comment by anurag-rajawat** [comment]
@rohanKanojia this PR is completed, but due to #1354 and #1356 there may be conflicts.
As this PR will remove hamcrest completely including its dependencies from `pom.xml` files. And if I exclude these 2 files then hamcrest still remains in the repository

Also, the above conflicts I'll resolve as these mentioned PRs get merged.

So, what should I do?

**Comment by rohanKanojia** [comment]
You can either wait or pick up some other issue to work on.

File: jkube-kit/build/service/docker/pom.xml [hunk 1]
```diff
@@ -76,13 +76,5 @@
...
       <artifactId>assertj-core</artifactId>
     </dependency>
-    <dependency>
-      <groupId>org.hamcrest</groupId>
-      <artifactId>hamcrest-all</artifactId>
-    </dependency>
-    <dependency>
-      <groupId>org.mockito</groupId>
-      <artifactId>mockito-core</artifactId>
-    </dependency>
```
Comment (rohanKanojia) [inline jkube-kit/build/service/docker/pom.xml:86]: Why is mockito dependency being removed?

File: jkube-kit/generator/api/src/test/java/org/eclipse/jkube/generator/api/support/AbstractPortsExtractorTest.java [hunk 2]
```diff
@@ -43,8 +42,8 @@ public void testReadConfigFromFile() {
                 ".properties",
                 "++suffix.yaml"}) {
             Map<String, Integer> map = extractFromFile("vertx.config", getClass().getSimpleName() + path);
-            assertThat(map, hasEntry("http.port", 80));
-            assertThat(map, hasEntry("https.port", 443));
+            assertThat(map).containsEntry("http.port", 80);
+            assertThat(map).containsEntry("https.port", 443);
```
Comment (rohanKanojia) [inline jkube-kit/generator/api/src/test/java/org/eclipse/jkube/generator/api/support/AbstractPortsExtractorTest.java:46]: chain these two assertions together

Comment (anurag-rajawat) [reply jkube-kit/generator/api/src/test/java/org/eclipse/jkube/generator/api/support/AbstractPortsExtractorTest.java:46]: Sure, will do it

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/DefaultServiceEnricherTest.java [hunk 3]
```diff
@@ -60,103 +60,136 @@ public void checkDefaultConfiguration() throws Exception {
         setupExpectations("type", "LoadBalancer");
 
         String json = enrich();
-        assertThat(json, isJson());
-        assertThat(json, hasJsonPath("$.spec.type", equalTo("LoadBalancer")));
+        ReadContext ctx = JsonPath.parse(json);
```
[the remark this answers is not in the document]
Comment (anurag-rajawat) [reply jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/DefaultServiceEnricherTest.java:63]: Yes, not only in this there are a couple of tests in `ImageEnricherTest.java`  and `DockerHealthCheckEnricherTest.java`

**Review by rohanKanojia — APPROVED** [review]
Looks good, Nice work :+1: Thanks!

**Comment by anurag-rajawat** [comment]
@manusa kindly review this PR

**Comment by anurag-rajawat** [comment]
@rohanKanojia is there anything to do?

**Review by manusa — APPROVED** [review]
LGTM, thx! :raised_hands:

**Pull request merged**